python/lsst/pipe/tasks/postprocess.py

```diff
@@ -78,6 +83,96 @@ def flattenFilters(df, noDupCols=["coord_ra", "coord_dec"], camelCase=False, inp
     return newDf
 
 
+class TableVStack:
+    """A helper class for stacking astropy tables without having them all in
+    memory at once.
+
+    Parameters
+    ----------
+    capacity : `int`
+        Full size of the final table.
+
+    Notes
+    -----
+    Unlike `astropy.table.vstack`, this class requires all tables to have the
+    exact same columns (it's slightly more strict than even the
+    ``join_type="exact"`` argument to `astropy.table.vstack`).
+    """
+
+    def __init__(self, capacity):
+        self.index = 0
+        self.capacity = capacity
+        self.result = None
+
+    @classmethod
+    def from_handles(cls, handles):
+        """Construct from an iterable of
+        `lsst.daf.butler.DeferredDatasetHandle`.
+
+        Parameters
+        ----------
+        handles : `~collections.abc.Iterable` [ \
+                `lsst.daf.butler.DeferredDatasetHandle` ]
+            Iterable of handles.   Must have a storage class that supports the
+            "rowcount" component, which is all that will be fetched.
+
+        Returns
+        -------
+        vstack : `TableVStack`
+            An instance of this class, initialized with capacity equal to the
+            sum of the rowcounts of all the given table handles.
+        """
+        capacity = sum(handle.get(component="rowcount") for handle in handles)
+        return cls(capacity=capacity)
+
+    def extend(self, table):
+        """Add a single table to the stack.
+
+        Parameters
+        ----------
+        table : `astropy.table.Table`
+            An astropy table instance.
+        """
+        if self.result is None:
+            self.result = astropy.table.Table()
+            for name in table.colnames:
+                column = table[name]
+                column_cls = type(column)
+                self.result[name] = column_cls.info.new_like([column], self.capacity, name=name)
+                self.result[name][:len(table)] = column
+            self.index = len(table)
+            self.result.meta = table.meta.copy()
+        else:
+            next_index = self.index + len(table)
+            for name in table.colnames:
+                self.result[name][self.index:next_index] = table[name]
```

Comment:
You promise "strict" in the docstring, but this doesn't actually check that the column names/types are the same (or extra columns).  Is this because it would be too inefficient and we just yolo?

Reply:
Yeah, I meant "strict" as in, "it's on you, the caller, to guarantee this, because I'm gonna assume it's the case."  But I don't think the checks would be that expensive, as long as it's just a numpy dtype comparison for each column block.  I'll add it.